libraries/resource_providers/api200/volume_provider.rb

```diff
@@ -18,11 +18,12 @@ def load_resource_with_associated_resources
         if @new_resource.volume_template
           @item.set_storage_volume_template(resource_named(:VolumeTemplate).new(@item.client, name: @new_resource.volume_template))
         else # Can't set the storage_pool or snapshot_pool if we specify a volume_template
-          load_storage_system if @new_resource.storage_system
+          validate_required_properties(:storage_system, :storage_pool)
+          storage_system_uri = load_storage_system
           # @item.set_storage_pool(resource_named(:StoragePool).new(@item.client, name: storage_pool)) if storage_pool
```

Comment:
I guess this was already fixed in Ruby SDK, check if this comment is still valid.